hcl2template/addrs/plugin.go

```diff
@@ -98,67 +134,63 @@ func IsPluginPartNormalized(str string) (bool, error) {
 //
 // The following are valid source string formats:
 //
-//	name
 //	namespace/name
 //	hostname/namespace/name
 func ParsePluginSourceString(str string) (*Plugin, hcl.Diagnostics) {
-	ret := &Plugin{
-		Hostname:  "",
-		Namespace: "",
-	}
 	var diags hcl.Diagnostics
 
-	// split the source string into individual components
-	parts := strings.Split(str, "/")
-	if len(parts) != 3 {
-		diags = diags.Append(&hcl.Diagnostic{
+	// Trim both leading and trailing slashes for convenience
+	str = strings.Trim(str, "/")
+
+	url, err := url.Parse(str)
+	if err != nil {
+		return nil, diags.Append(&hcl.Diagnostic{
 			Severity: hcl.DiagError,
-			Summary:  "Invalid plugin source string",
-			Detail:   `The "source" attribute must be in the format "hostname/namespace/name"`,
+			Summary:  "Invalid source URL",
+			Detail:   "A source must be a valid URL",
```

Comment:
Yep without a scheme as I believe we do check it? Even without scheme this may be a valid URL, but I agree the error message can be improved, I'll see what I can reroll there